app/controllers/todos_controller.rb

```diff
@@ -16,7 +19,9 @@ def new
 
   def create
     @todo = Todo.new(todo_params)
+    @todo.user = current_user
     @todo.duedate = Time.new(params[:year], params[:month], params[:day])
+    @todo.category = Category.find_by(name: params[:category][:name])

```

Comment:
This line seems to throw an exception when form is submitted without a category:

![image](https://user-images.githubusercontent.com/21155293/30892636-b560ce8c-a307-11e7-8dae-f906476bb2b0.png)

Reply:
Just realized our mistake. Nice catch.

Reply:
Made a hotfix, should work now